src/cueUtilities.js

```diff
@@ -196,139 +237,213 @@ module.exports = function (params, cy, api) {
-      var opts = options();
-			if (node){
-				var expandcollapseRenderedStartX = node._private.data.expandcollapseRenderedStartX;
-				var expandcollapseRenderedStartY = node._private.data.expandcollapseRenderedStartY;
-				var expandcollapseRenderedRectSize = node._private.data.expandcollapseRenderedCueSize;
-				var expandcollapseRenderedEndX = expandcollapseRenderedStartX + expandcollapseRenderedRectSize;
-				var expandcollapseRenderedEndY = expandcollapseRenderedStartY + expandcollapseRenderedRectSize;
-                
-                var cyRenderedPos = event.renderedPosition || event.cyRenderedPosition;
-				var cyRenderedPosX = cyRenderedPos.x;
-				var cyRenderedPosY = cyRenderedPos.y;
-				var factor = (opts.expandCollapseCueSensitivity - 1) / 2;
-
-				if ( (Math.abs(oldMousePos.x - currMousePos.x) < 5 && Math.abs(oldMousePos.y - currMousePos.y) < 5)
-					&& cyRenderedPosX >= expandcollapseRenderedStartX - expandcollapseRenderedRectSize * factor
-					&& cyRenderedPosX <= expandcollapseRenderedEndX + expandcollapseRenderedRectSize * factor
-					&& cyRenderedPosY >= expandcollapseRenderedStartY - expandcollapseRenderedRectSize * factor
-					&& cyRenderedPosY <= expandcollapseRenderedEndY + expandcollapseRenderedRectSize * factor) {
-					if(opts.undoable && !ur)
-						ur = cy.undoRedo({
-							defaultActions: false
-						});
-					if(api.isCollapsible(node))
-						if (opts.undoable){
-							ur.do("collapse", {
-								nodes: node,
-								options: opts
-							});
-						}
-						else
-							api.collapse(node, opts);
-				else if(api.isExpandable(node))
-					if (opts.undoable)
-						ur.do("expand", {
-							nodes: node,
-							options: opts
-						});
-					else
-						api.expand(node, opts);
-					}
-			}
-		});
+        var oldMousePos = null, currMousePos = null;
+        cy.on('mousedown', 'node', data.eMouseDown = function(e){
+          oldMousePos = e.renderedPosition || e.cyRenderedPosition
+        });
+        cy.on('mouseup', 'node', data.eMouseUp = function(e){
+          currMousePos = e.renderedPosition || e.cyRenderedPosition
+        });
+
+        cy.on('grab', 'node', data.eGrab = function (e) {
+          preventDrawing = true;
+        });
+
+        cy.on('free', 'node', data.eFree = function (e) {
+          preventDrawing = false;
+        });
+
+        cy.on('position', 'node', data.ePosition = function () {
+          if (hoveredGroup) {
+            clearDraws();
+            if (options().appearOnGroupSelect) {
+              drawCuesForSelectedGroups();
+            }
+          }
+        });
+
+        cy.on('remove', 'node', data.eRemove = function () {
+          if (!options().appearOnGroupSelect) {
+            clearDraws();
+            hoveredGroup = null;
+          }
+        });
+
+        var ur;
+        cy.on('select', 'node', data.eSelect = function(evt){
+          if (this.length > cy.nodes(":selected").length)
+            this.unselect();
```

Comment:
I am actually not so sure, this.length gets the length of a node in this case, which should always be one..
cy.nodes(":selected") gets selected nodes I believe. I don't know why it does this, I just left this code alone in case something breaks.